library/akamai.py

```diff
@@ -1,29 +1,130 @@
-#!/usr/bin/env python
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+"""
+Provides authentication using local `.edgerc` files for Akamai API calls.
+Note this replaces the ansible.builtin.uri module which will not do
+Akamai authentication with `.edgerc` files.
+"""
+
+# Copyright: (c) 2018, Jacob Hudson <jacob.alan.hudson@gmail.com>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
 
-from ansible.module_utils.basic import *
-import requests
 import json
 
+# import urllib.parse
+from urllib.parse import urljoin
+from os.path import expanduser
+
+import requests
+
 try:
     from akamai.edgegrid import EdgeGridAuth, EdgeRc
 except ImportError:
-    print "Please install `edgegrid-python` using pip"
-from os.path import expanduser
-from urlparse import urljoin
+    print("Please install `edgegrid-python` using pip")
+
+from ansible.module_utils.basic import AnsibleModule
+
+DOCUMENTATION = r'''
+---
+module: ansible-akamai
+short_description: Ansible module for working with Akamai OPEN APIs
+description:
+    - Ansible Module for working with Akamai OPEN APIs
+    - Provides .edgerc Akamai API authentication which is not available in ansible.builtin.uri
+    - For more information, see https://github.com/akamai/AkamaiOPEN-edgegrid-python
+      and https://techdocs.akamai.com/home/page/products-tools-a-z?sort=api
+author: "Jacob Hudson (@jacob-hudson)"
+options:
+    section:
+        description:
+            - Section header of .edgerc ini file.
+            - .edgerc file described in more detail at
+              https://techdocs.akamai.com/developer/docs/set-up-authentication-credentials.
+        required: true
+        default: null
+        choices: []
+        aliases: []
+        version_added: null
+    endpoint:
+        description:
+            - Trailing API URI path after the API baseurl in format `https://host`.
+            - Needs to include the starting slash.
+            - Provides a re-usable and inspecific way to access any Akamai API endpoint.
+        required: true
+    method:
+        description:
+            - HTTP method to use for the API request.
+            - Provides a re-usable and inspecific way to access any Akamai API endpoint.
+        required: true
+        default: null
+        choices:
+          - GET
+          - POST
+          - PUT
+        aliases: []
+        version_added: null
+    body:
+        description:
+            - The request body that needs to used only for POST method.
+            - This needs to be path to a JSON-formatted file.
+        required: true
+        default: null
+        choices: []
+        aliases: []
+        version_added: null
+    headers:
+        description:
+            - The request headers that needs to used only for POST method
+            - Must be written in sentences.
+        required: true
+        default:
+          - Content-Type: application/json
```

Comment:
It appears there is some inconsistent spacing here.

Reply:
Yep, pretty much copy/paste from [here](https://github.com/ansible/ansible/blob/devel/examples/DOCUMENTATION.yml).